modules/atomic-widgets/base/atomic-element-base.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\Base;
+
+use Elementor\Element_Base;
+use Elementor\Modules\AtomicWidgets\Controls\Section;
+use Elementor\Modules\AtomicWidgets\PropTypes\Concerns\Has_Atomic_Base;
+use Elementor\Modules\AtomicWidgets\Styles\Style_Schema;
+use Elementor\Modules\AtomicWidgets\Validators\Props_Validator;
+use Elementor\Modules\AtomicWidgets\PropTypes\Contracts\Prop_Type;
+use Elementor\Modules\AtomicWidgets\Validators\Styles_Validator;
+use Elementor\Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+abstract class Atomic_Element_Base extends Element_Base {
+
+	use Has_Atomic_Base;
+
+	protected $version = '0.0';
+	protected $styles = [];
+
+	public function __construct( $data = [], $args = null ) {
+		parent::__construct( $data, $args );
+
+		$this->version = $data['version'] ?? '0.0';
+		$this->styles = $data['styles'] ?? [];
+	}
+
+	private function get_valid_controls( array $schema, array $controls ): array {
```

Comment:
@nevo-elementor @Omerisra6 Some of this code looks duplicated from the widget base. How do you want to handle it?

Reply:
Maybe we could create a trait for the common logic?